csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -209,14 +268,13 @@ private void WriteMessage(TextWriter writer, IMessage message)
                 }
             }
             writer.Write("{ ");
-            bool writtenFields = WriteMessageFields(writer, message, false);
-            writer.Write(writtenFields ? " }" : "}");
+            bool nothingWritten = WriteMessageFields(writer, message, true);
+            writer.Write(nothingWritten ? "}" : " }");
         }
 
-        private bool WriteMessageFields(TextWriter writer, IMessage message, bool assumeFirstFieldWritten)
+        private bool WriteMessageFields(TextWriter writer, IMessage message, bool firstField)
```

Comment:
Okay - if you do that, I'll then add more documentation to make the motivation clearer in a later commit. Thanks!